examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -224,33 +247,44 @@ export class TaskList extends DataObject implements ITaskList {
 
         // TODO: Delete any items that are in the root but missing from the external data
         const updateTaskPs = updatedExternalData.map(async ([id, { name, priority }]) => {
-            const currentTask = this.draftData.get<PersistedTask>(id);
             // Write external data into savedData map.
-            this.savedData.set(id, currentTask);
-
+            const currentTask = this.savedData.get<PersistedTask>(id);
+            // Create a new task because it doesn't exist already
             if (currentTask === undefined) {
+                const savedNameString = SharedString.create(this.runtime);
```

Comment:
Option for a future PR (not this one) -- the saved data doesn't need to be a sharedstring/sharedcell since we're not permitting live edits on it.  We could save the data directly into the savedData map.